content/StsServerIdentity/Startup.cs

```diff
@@ -201,38 +207,27 @@ public void Configure(IApplicationBuilder app)
             });
         }
 
-        private static X509Certificate2 GetCertificate(IWebHostEnvironment environment, IConfiguration configuration)
+        private static (X509Certificate2, X509Certificate2) GetCertificates(IWebHostEnvironment environment, IConfiguration configuration)
         {
-            X509Certificate2 cert;
-            var useLocalCertStore = Convert.ToBoolean(configuration["UseLocalCertStore"]);
-            var certificateThumbprint = configuration["CertificateThumbprint"];
-
-            if (environment.IsProduction())
-            {
-                if (useLocalCertStore)
-                {
-                    using (X509Store store = new X509Store(StoreName.My, StoreLocation.LocalMachine))
-                    {
-                        store.Open(OpenFlags.ReadOnly);
-                        var certs = store.Certificates.Find(X509FindType.FindByThumbprint, certificateThumbprint, false);
-                        cert = certs[0];
-                        store.Close();
-                    }
-                }
-                else
-                {
-                    // Azure deployment, will be used if deployed to Azure
-                    var vaultConfigSection = configuration.GetSection("Vault");
-                    var keyVaultService = new KeyVaultCertificateService(vaultConfigSection["Url"], vaultConfigSection["ClientId"], vaultConfigSection["ClientSecret"]);
-                    cert = keyVaultService.GetCertificateFromKeyVault(vaultConfigSection["CertificateName"]);
-                }
-            }
-            else
+            var certificateConfiguration = new CertificateConfiguration
             {
-                cert = new X509Certificate2(Path.Combine(environment.ContentRootPath, "sts_dev_cert.pfx"), "1234");
-            }
+                // Use an Azure key vault
+                CertificateNameKeyVault = configuration["CertificateNameKeyVault"], //"StsCert",
+                KeyVaultEndpoint = configuration["AzureKeyVaultEndpoint"], // "https://damienbod.vault.azure.net"
+
+                // development certificate
+                DevelopmentCertificatePfx = Path.Combine(environment.ContentRootPath, "sts_dev_cert.pfx"),
+                DevelopmentCertificatePassword = "1234" //configuration["DevelopmentCertificatePassword"] //"1234",
+
+                //    // Use a local store with thumbprint
+                //    UseLocalCertStore = Convert.ToBoolean(configuration["UseLocalCertStore"]),
+                //    CertificateThumbprint = configuration["CertificateThumbprint"],

```

Comment:
Remove dead code ;-) and comments above that are not used...the just confuse :-D

Reply:
This is my documentation for usage with server deployments not using key vault